pkg/machine/e2e/list_test.go

```diff
@@ -70,6 +70,34 @@ var _ = Describe("podman machine list", func() {
 		Expect(util.StringInSlice(name1, listNames)).To(BeTrue())
 		Expect(util.StringInSlice(name2, listNames)).To(BeTrue())
 	})
+
+	It("list machine: check if running while starting", func() {
+		i := new(initMachine)
+		session, err := mb.setCmd(i.withImagePath(mb.imagePath)).run()
+		Expect(err).To(BeNil())
+		Expect(session.ExitCode()).To(Equal(0))
+
+		s := new(startMachine)
+		startSession, err := mb.setCmd(s).runWithoutWait()
+		Expect(err).To(BeNil())
+		l := new(listMachine)
+		for { // need to be infinite because we need to check AFTER the loop.
```

Comment:
would this loop still timeout at some max test time?

Reply:
yes because `startSession` will time out after 90 seconds and `ExitCode()` will never be -1 in that case